core/src/main/scala/org/apache/spark/api/r/SerDe.scala

```diff
@@ -355,6 +355,13 @@ private[spark] object SerDe {
           writeInt(dos, v.length)
           v.foreach(elem => writeObject(dos, elem))
 
+        // Handle Properties
```

Comment:
Yeah it still feels awkward to just do this specially for the Properties object. 

@felixcheung Do you have an idea what part of the code would move to scala if we want to do it on the scala side ? Typically we do deal with such conversions on the scala side, so thats the main reason I'm asking. Is it just the `varargsToJProperties` function ?